src/change_stream.ts

```diff
@@ -683,15 +683,15 @@ function processNewChange<TSchema>(
   callback?: Callback<ChangeStreamDocument<TSchema>>
 ) {
   if (changeStream[kClosed]) {
-    // TODO(NODE-3405): Replace with MongoStreamClosedError
-    if (callback) callback(new MongoDriverError(CHANGESTREAM_CLOSED_ERROR));
+    // TODO(NODE-3485): Replace with MongoChangeStreamStreamClosedError
+    if (callback) callback(new MongoAPIError(CHANGESTREAM_CLOSED_ERROR));
     return;
   }
 
   // a null change means the cursor has been notified, implicitly closing the change stream
   if (change == null) {
-    // TODO(NODE-3405): Replace with MongoStreamClosedError
-    return closeWithError(changeStream, new MongoDriverError(CHANGESTREAM_CLOSED_ERROR), callback);
+    // TODO(NODE-3485): Replace with MongoChangeStreamStreamClosedError
+    return closeWithError(changeStream, new MongoAPIError(CHANGESTREAM_CLOSED_ERROR), callback);
```

Comment:
Yes, they're all API errors. They catch the case where someone has closed their ChangeStream before we are able to execute the logic beneath these checks. As an example, processing the resume queue will fail with the API error because the ChangeStream has been closed. So it does seem like incorrect usage.

Reply:
Sorry Andy I overlooked this one when we were on the call, you're correct. Except for this one, this is the one case where the `change` variable is null because the underlying cursor gave us null. Since change streams are "forever" this would only happen if there was some server reason for shutting the change stream so this would not be the fault of usage but rather a runtime issue.

Reply:
This specific one isn't an api error, also, could update all the TODO comments in this file to remove the duplicate `StreamStream` spelling?

Reply:
Oops! Going to change this to be MongoRuntimeError and leave it as such. I think creating an entirely new error for this might be overkill.